source/slang/lower-to-ir.cpp

```diff
@@ -5790,24 +5790,22 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
         // body appear before the function itself in the list
         // of global values.
         irFunc->moveToEnd();
-
         return LoweredValInfo::simple(finishOuterGenerics(subBuilder, irFunc));
     }
 
     LoweredValInfo visitGenericDecl(GenericDecl * genDecl)
     {
         // TODO: Should this just always visit/lower the inner decl?
-
         if (auto innerFuncDecl = genDecl->inner->As<FunctionDeclBase>())
-            return lowerFuncDecl(innerFuncDecl);
+            return ensureDecl(context, innerFuncDecl);
```

Comment:
It might be out of scope for this PR, but it sure looks like we could change the body of `visitGenericDecl` to just be `return ensureDecl(context, genDecl->inner)`, since all the implemented cases pretty much do that. The one exception is the `StructDecl` case, and I'm not actually clear on why it returns `LoweredValInfo()` in that case.